src/AzureExtension/DataManager/AzureDataManagerCache.cs

```diff
@@ -0,0 +1,316 @@
+        try
+        {
+            developerIds = DeveloperIdProvider.GetInstance().GetLoggedInDeveloperIdsInternal();
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Failed getting logged in developer ids.");
+            firstException = ex;
+            ++errors;
+            developerIds = [];
+        }
+
+        foreach (var developerId in developerIds)
+        {
+            _log.Debug($"Updating accounts for {developerId.LoginId}, older than {olderThan}");
+            var accounts = GetAccounts(developerId, cancellationToken);
+            foreach (var account in accounts)
+            {
+                var org = Organization.Get(DataStore, account.AccountUri.ToString());
+                if (org is not null)
+                {
+                    if ((DateTime.UtcNow - org.LastSyncAt) < olderThan)
+                    {
+                        _log.Debug($"Organization: {org.Name} has recently been updated, skipping.");
+                        ++accountsSkipped;
+                        continue;
+                    }
+                }
+
+                // Transactional unit is the account (organization), which chains down to Project
+                // and repository.
+                using var tx = DataStore.Connection!.BeginTransaction();
+                try
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    var connection = AzureClientProvider.GetConnectionForLoggedInDeveloper(account.AccountUri, developerId);
+
+                    // If connection is null the organization is disabled
+                    if (connection is null)
+                    {
+                        ++accountsSkipped;
+                        _log.Information($"Account {account.AccountName} had a null connection, treating as disabled.");
+                        tx.Rollback();
+                        continue;
+                    }
+
+                    UpdateOrganization(account, developerId, connection, cancellationToken);
+
+                    tx.Commit();
+                    _log.Information($"Updated organization: {account.AccountName}");
+                    ++accountsUpdated;
+                }
+                catch (OperationCanceledException cancelException)
+                {
+                    firstException ??= cancelException;
+                    tx.Rollback();
+                    _log.Information("Operation was cancelled.");
+                    dynamic cancelContext = new ExpandoObject();
+                    var cancelContextDict = (IDictionary<string, object>)cancelContext;
+                    cancelContextDict.Add("AccountsUpdated", accountsUpdated);
+                    cancelContextDict.Add("AccountsSkipped", accountsSkipped);
+                    cancelContextDict.Add("Errors", errors);
+                    cancelContextDict.Add("TimeElapsed", DateTime.UtcNow - startTime);
```

Comment:
A new method to save some lines in two places
```suggestion
                    dynamic context = CreateUpdateSummary(accountsUpdated, accountsSkipped, errors, DateTime.UtcNow - startTime);
```

Reply:
Added and fixed